airbyte-integrations/connectors/source-shopify/unit_tests/graphql_bulk/test_job.py

```diff
@@ -0,0 +1,173 @@
+
+
+@pytest.mark.parametrize(
+    "bulk_job_response, expected",
+    [
+        ("bulk_error_with_concurrent_job", True),
+        ("bulk_successful_response", False),
+    ],
+)
+def test_has_running_concurrent_job(request, requests_mock, bulk_job_response, auth_config, expected) -> None:
+    stream = MetafieldOrders(auth_config)
+    requests_mock.get(stream.graphql_path, json=request.getfixturevalue(bulk_job_response))
+    test_response = requests.get(stream.graphql_path)
+    assert stream.bulk_job.has_running_concurrent_job(test_response) == expected
+
+
+@pytest.mark.parametrize(
+    "bulk_job_response, expected",
+    [
+        ("bulk_error", "[{'field': 'some_field', 'message': 'something wrong with the requested field.'}]"),
+    ],
+)
+def test_job_check_for_errors(request, requests_mock, bulk_job_response, auth_config, expected) -> None:
+    stream = MetafieldOrders(auth_config)
+    requests_mock.get(stream.graphql_path, json=request.getfixturevalue(bulk_job_response))
+    test_response = requests.get(stream.graphql_path)
+    with pytest.raises(ShopifyBulkExceptions.BulkJobError) as error:
+        stream.bulk_job.job_check_for_errors(test_response)
+    assert expected in repr(error.value)
+
+
+@pytest.mark.parametrize(
+    "bulk_job_response, expected",
+    [("bulk_successful_response", "gid://shopify/BulkOperation/4046733967549"), ("bulk_error", None)],
+)
+def test_job_get_id(request, requests_mock, bulk_job_response, auth_config, expected) -> None:
+    stream = MetafieldOrders(auth_config)
+    requests_mock.get(stream.graphql_path, json=request.getfixturevalue(bulk_job_response))
+    test_response = requests.get(stream.graphql_path)
+    assert stream.bulk_job.job_get_id(test_response) == expected
+
+
+@pytest.mark.parametrize(
+    "bulk_job_response, error_type, expected",
+    [
+        ("bulk_successful_response", None, "gid://shopify/BulkOperation/4046733967549"),
+        (
+            "bulk_error",
+            ShopifyBulkExceptions.BulkJobError,
+            "[{'field': 'some_field', 'message': 'something wrong with the requested field.'}]",
+        ),
+        ("bulk_error_with_concurrent_job", None, None),
+    ],
+)
+def test_job_create(request, requests_mock, bulk_job_response, auth_config, error_type, expected) -> None:
+    stream = MetafieldOrders(auth_config)
+    # patching concurent settings
+    stream.bulk_job.concurrent_max_retry = 1  # 1 attempt max
+    stream.bulk_job.concurrent_interval_sec = 1  # 1 sec
+    #
+    requests_mock.get(stream.graphql_path, json=request.getfixturevalue(bulk_job_response))
+    if error_type:
+        with pytest.raises(error_type) as error:
+            stream.bulk_job.job_create(requests.get(stream.graphql_path).request)
+        assert expected in repr(error.value)
+    else:
+        result = stream.bulk_job.job_create(requests.get(stream.graphql_path).request)
+        assert stream.bulk_job.job_get_id(result) == expected
+
+
+@pytest.mark.parametrize(
```

Comment:
can you add a case for a non-error response that does not have a job_id to cover [the case of a Non job id](https://github.com/airbytehq/airbyte/pull/32345/files#r1446602108)?